ixmp/core.py

```diff
@@ -595,6 +596,101 @@ def remove_timeseries(self, df):
                 years.add(java.Integer(y))
             self._jobj.removeTimeseries(name[0], name[1], None, years, name[2])
 
+    def add_geodata(self, df):
+        """Add geodata (layers) to the TimeSeries.
+
+        Parameters
+        ----------
+        df : :class:`pandas.DataFrame`
+            Data to add. `df` must have the following columns:
+
+            - `region`
+            - `variable`
+            - `time`
+            - `unit`
+            - `year`
+            - `value`
+            - `meta`
+        """
+        for i in df.index:
+            self._jobj.addGeoData(df.region[i],
+                                  df.variable[i],
+                                  df.time[i],
+                                  java.Integer(int(df.year[i])),
+                                  df.value[i],
+                                  df.unit[i],
+                                  int(df.meta[i]))
+
+    def remove_geodata(self, df):
+        """Remove geodata from the TimeSeries instance.
+
+        Parameters
+        ----------
+        df : :class:`pandas.DataFrame`
+            Data to remove. `df` must have the following columns:
+
+            - `region`
+            - `variable`
+            - `unit`
+            - `time`
+            - `year`
+        """
+        for values, _df in df.groupby(['region', 'variable', 'time', 'unit']):
+            years = java.LinkedList()
+            for year in _df['year']:
+                years.add(java.Integer(year))
+            self._jobj.removeGeoData(values[0], values[1], values[2], years,
+                                     values[3])
+
+    def _get_timespans(self):
+        self._timespans = {}
+        timespan_cls = JClass('at.ac.iiasa.ixmp.objects.TimeSeries$TimeSpan')
+        for timespan in timespan_cls.values():
+            self._timespans[timespan.ordinal()] = timespan.name()
+
+    def _get_timespan_name(self, ordinal):
```

Comment:
would it make sense to change java API to use strings instead of enums if JPype doesn't provide convenient way to work with enums?